kin/account.py

```diff
@@ -139,142 +141,128 @@ def get_transaction_history(self, amount=10, descending=True, cursor=None, simpl
+        :raises: :class:`KinErrors.NotValidParamError`: if the memo is longer than MEMO_CAP characters
+        :raises: KinErrors.NotValidParamError: if the amount is too precise
+        :raises: KinErrors.NotValidParamError: if the fee is not an too precise
         """
         if not is_valid_address(address):
-            raise ValueError('invalid address: {}'.format(address))
+            raise KinErrors.StellarAddressInvalidError('invalid address: {}'.format(address))
+
+        if float(starting_balance) < 0:
+            raise ValueError('Starting balance : {} cant be negative'.format(starting_balance))
 
         # Build the transaction and send it.
 
         builder = self.channel_manager.build_transaction(lambda builder:
                                                          partial(builder.append_create_account_op, address,
                                                                  str(starting_balance)),
+                                                         fee,
                                                          memo_text=build_memo(self.app_id, memo_text))
         return Transaction(builder, self.channel_manager)
 
-    def build_send_xlm(self, address, amount, memo_text=None):
-        """Send XLM to the account identified by the provided address.
+    def build_send_kin(self, address, amount, fee, memo_text=None):
+        """Build a tx to send asset to the account identified by the provided address.
 
-        :param str address: the account to send XLM to.
+        :param str address: the account to send asset to.
 
-        :param number amount: the number of XLM to send.
+        :param number amount: the asset amount to send.
 
-        # TODO: might want to limit this if we do tx coloring
         :param str memo_text: (optional) a text to put into transaction memo.
 
+        :param float fee: fee to be deducted for the tx
+
         :return: a transaction object
         :rtype: :class: `Kin.Transaction`
 
         :raises: ValueError: if the provided address has a wrong format.
         :raises: ValueError: if the amount is not positive.
+        :raises: KinErrors.NotValidParamError: if the amount is too precise
+        :raises: KinErrors.NotValidParamError: if the fee is not an too precise
         """
-        return self._build_send_asset(Asset.native(), address, amount, memo_text)
-
-    def build_send_kin(self, address, amount, memo_text=None):
-        """Send KIN to the account identified by the provided address.
 
-         :param str address: the account to send KIN to.
-
-         :param number amount: the amount of KIN to send.
+        if not is_valid_address(address):
+            raise KinErrors.StellarAddressInvalidError('invalid address: {}'.format(address))
 
-         # TODO: might want to limit this if we do tx coloring
-         :param str memo_text: (optional) a text to put into transaction memo.
+        if float(amount) <= 0:
```

Comment:
Used float since I thought we will specify fees with Kin, I will change it.

We dont do any math with the numbers we get in the kin-sdk, only checking that its negative or positive.

The calculation happen the kin-base sdk, which uses strings and decimals for it